yt/frontends/parthenon/data_structures.py

```diff
@@ -17,7 +17,9 @@
 from .fields import ParthenonFieldInfo
 
 geom_map = {
-    "UniformCartesian": Geometry.CARTESIAN,
+    "UniformCartesian": (Geometry.CARTESIAN,("x","y","z")),
+    "UniformCylindrical": (Geometry.CYLINDRICAL,("r","theta","z")),
```

Comment:
@forrestglines I suspect this line is the only reason for your latest commit, so I should point out that you should just use `Geometry.POLAR` here, so you don't need to specify axis_order at all.

Reply:
Thanks @neutrinoceros that would certainly allow a bit of code cleanup. Unfortunately changing to `Geometry.POLAR` is breaking my scripts which use fixed resolution buffers. For example, this code to make a slice plot works with both `Geometry.POLAR` and the current solution with `Geometry.CYLINDRICAL`:
```python
ds = yt.load(filename)

slc = yt.SlicePlot(ds, "z", ("gas", "density"))
```
but this code below only works with my current solution with `Geometry.CYLINDRICAL`:
```python
ds = yt.load(filename)

slc = yt.SlicePlot(ds, "z", ("gas", "density"))
frb = slc.data_source.to_frb(width=(4,"cm"),resolution=512,center=(0,0))
frb["r"] 
```
with an error being thrown by `pixelize_cylinder`
```
File ~/installs-chicoma/miniconda3/envs/py39/lib/python3.9/site-packages/yt/visualization/fixed_resolution.py:600, in CylindricalFixedResolutionBuffer._generate_image_and_mask(self, item)
    597 @override
    598 def _generate_image_and_mask(self, item) -> None:
    599     buff = np.zeros(self.buff_size, dtype="f8")
--> 600     mask = pixelize_cylinder(
    601         buff,
    602         self.data_source["r"],
    603         self.data_source["dr"],
    604         self.data_source["theta"],
    605         self.data_source["dtheta"],
    606         self.data_source[item].astype("float64"),
    607         self.radius,
    608         return_mask=True,
    609     )
    610     self.data[item] = ImageArray(
    611         buff, units=self.data_source[item].units, info=self._get_info(item)
    612     )
    613     self.mask[item] = mask

File ~/installs-chicoma/miniconda3/envs/py39/lib/python3.9/site-packages/yt/utilities/lib/pixelization_routines.pyx:552, in yt.utilities.lib.pixelization_routines.pixelize_cylinder()

ValueError: Buffer has wrong number of dimensions (expected 2, got 1)
```

Suggestions? Is this something to be fixed in the frontend or in yt? I can provide the data file for this cylindrical dataset. The dataset will be included with our tests

Reply:
Hard to tell for sure without the data file, but I can have a look if you can share it.

Reply:
Thanks! I've pushed the small changes to switch from `Geometry.CYLINDRICAL` to `Geometry.POLAR`.

Here's the file I was testing on (with an .txt extension due to github's limitations)

[disk_cyl.prim.00000.phdf.txt](https://github.com/yt-project/yt/files/14058885/disk_cyl.prim.00000.phdf.txt)

Reply:
thank you ! I can reproduce locally, which is a great start. I'll try to debug this today.

Reply:
Actually it's a yt bug, I can also reproduce it with AMRVAC data
```python
import yt

ds = yt.load_sample("bw_polar_2d")
slc = yt.SlicePlot(ds, "z", ("gas", "density"))
frb = slc.data_source.to_frb(width=(4, "cm"), resolution=512, center=(0, 0))
frb["r"]
```
I'll open a dedicated issue.

Reply:
I saw issue #4790, the bug is not related to 2D datasets. I also explored this with a 3D dataset and got the same error.

[disk_cyl.3d.prim.00000.phdf.txt](https://github.com/yt-project/yt/files/14073840/disk_cyl.3d.prim.00000.phdf.txt)

Reply:
Fantastic that we have a bug fix started with this so soon. While this did eliminate the error with FRB, the polar coordinates are still not behaving like the cylindrical coordinates. In some comparison plots between Athena++ and AthenaPK using FRB's of a slice generated with the same code, I get plots looking like this:

![Screen Shot 2024-01-27 at 3 13 50 PM](https://github.com/yt-project/yt/assets/5799640/a3aab021-6450-4a44-9a20-1ea879e5b4a0)

My Parthenon frontend (using `Geometry.POLAR` with your new bugfix) is only getting one quadrant while the Athena++ frontend (using `Geometry.CYLINDRICAL`) is getting the full disk as expected.

Let me work on a better reproducer and perhaps one using an existing frontend in YT. I'm not convinced either way that this is now the Parthenon frontend's bug or Polar's bug but it's still a problem that changing from cylindrical to polar changes my plots.